staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresourcedefinition/strategy.go

```diff
@@ -222,3 +229,104 @@ func MatchCustomResourceDefinition(label labels.Selector, field fields.Selector)
 func CustomResourceDefinitionToSelectableFields(obj *apiextensions.CustomResourceDefinition) fields.Set {
 	return generic.ObjectMetaFieldsSet(&obj.ObjectMeta, true)
 }
+
+// dropDisabledFields drops disabled fields that are not used if their associated feature gates
+// are not enabled.
+func dropDisabledFields(newCRD *apiextensions.CustomResourceDefinition, oldCRD *apiextensions.CustomResourceDefinition) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.CustomResourceValidationExpressions) && !xValidationsInUse(oldCRD) {
+		if newCRD.Spec.Validation != nil && newCRD.Spec.Validation.OpenAPIV3Schema != nil {
+			dropXValidationsField(newCRD.Spec.Validation.OpenAPIV3Schema)
```

Comment:
we also have the per-version validations.

Reply:
Added pre-version validations